packages/data-broker/src/__tests__/services/dhis/DhisService/testPull/testPullAnalytics.js

```diff
@@ -225,6 +225,27 @@ export const testPullAnalytics = () => {
         expect(dhisApi.getEventAnalytics).toHaveBeenCalledTimes(2);
       });
 
+      it('invokes once for each programCode that has a dataElement associated', async () => {
+        await dhisService.pull([DATA_SOURCES.POP01, DATA_SOURCES.POP02], 'dataElement', {
+          programCodes: ['POP01', 'DIFF_GROUP', 'CODE_WITH_NO_ELEMENTS'],
+          useDeprecatedApi: false,
+        });
+        expect(dhisApi.getEventAnalytics).toHaveBeenCalledWith(
+          expect.objectContaining({
+            programCode: 'POP01',
+          }),
+        );
+        expect(dhisApi.getEventAnalytics).toHaveBeenCalledWith(
+          expect.objectContaining({
+            programCode: 'DIFF_GROUP',
+          }),
+        );
+
+        // Note: The api should NOT be called with programCode: CODE_WITH_NO_ELEMENTS
```

Comment:
Note to self: also could potentially hide errors?

Reply:
Taking a what would Kostas do approach, let's do what makes the code most closely match our intuitive separated concerns. Namely, that regardless of dataSource, you should only be able to pull for programCodes which are in the database. Making the logic part of the dhisApi doesn't make much sense as I'd expect that we want this functionality for all dataSources. So option 2 🎉

Reply:
Ended up only doing it in DhisService, as programCodes are only used in DHIS (which I didn't know)

Reply:
I guess after https://github.com/beyondessential/tupaia-backlog/issues/663 was implemented all elements are grouped under a data group, so we could use data groups/elements (and not surveys/questions)  for the `TupaiaService` too.

But we currently don't, so good call!